### ISSUE TITLE
Anafast crashes kernel in jupyter notebook

### ISSUE THREAD

**Issue by ianmgull** [healpy/healpy#683]
I've found that running anafast causes my notebook's kernel to die. This seems to happen regardless of my choice of NSIDE or LMAX. 

When running a cell with anafast, there is no python output. Only a  pop-up appears which says: 

"The kernel appears to have died. It will restart automatically"

After clicking ok, the notebook seems frozen until I restart the kernel.

**Comment by zonca** [comment]
what OS? how did you install healpy? what version?
please provide shortest snippet of code that makes it crash

**Comment by ianmgull** [comment]
I'm running Apple's Big Sur version 11.2.3. I installed healpy using:`pip install healpy` in my terminal. 

The following code in an empty Jupyter notebook cell produces the same result for me: 

```
import numpy as np
import healpy as hp

NSIDE = 256
LMAX = 400

test_array=np.ones(786432)
hp.anafast(test_array, lmax=LMAX)
```

**Comment by lpsinger** [comment]
Are you on an Apple Silicon machine?

**Comment by ianmgull** [comment]
Yes, I'm running one of the newer M1 Processors. 

Also, I've found that the short code snippet from my previous comment also causes issues outside of a jupyter notebook. If I save the above code as a .py file, and run it from an ipython terminal, the following message appears. 

> OMP: Error #15: Initializing libomp.dylib, but found libomp.dylib already initialized.
OMP: Hint This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results.

I'm leaning towards this being a Mac issue more than a healpy issue.

**Comment by lpsinger** [comment]
Well, it's kind of both. We don't have prebuilt wheels for M1 Macs yet, so it's being built locally on your machine.

**Comment by ianmgull** [comment]
> Well, it's kind of both. We don't have prebuilt wheels for M1 Macs yet, so it's being built locally on your machine.

Is this something that will be addressed in a future update, or is there something I could do on my machine to remedy the issue? 

Thanks!

**Comment by lpsinger** [comment]
What Python runtime are you using?

Is your Python runtime an x86_64 build running under Rosetta 2 emulation, or is it a native arm64 build?

**Comment by ianmgull** [comment]
I haven't explicitly set up a Rosetta emulation, so I assume it's a native build but I'm not entirely sure how to verify that.

**Comment by lpsinger** [comment]
No, Rosetta 2 is built in to macOS. It is what allows you to transparently run any x86_64 binary on your M1 Mac. What does this print?

```
python -c 'import platform; print(platform.platform())'
````

**Comment by ianmgull** [comment]
Thanks for the help. The output was: 

`macOS-10.16-x86_64-i386-64bit`

**Comment by ianmgull** [comment]
Hi @lpsinger, does the above output give any indication of what I might need to do to remedy the issue? 

Thanks

**Comment by ianmgull** [comment]
Does anyone know if the newest update addresses issues with M1 processors? 

Thanks!

**Comment by wangshiyuan911** [comment]
How was your question? I have difficulty when I tried to install healpy on my macM1. Do you know how to install it? Thank you!
